File: 系列3-分布式思维及源码分析/31-Rocketmq特性详解-悟空/rocketmq-externals-master/rocketmq-php/example/simple/Producer.php
<?php
/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
use RocketMQ\Client\Producer\DefaultMQProducer;
use RocketMQ\Common\Message\Message;

$producer = new DefaultMQProducer("ProducerGroupName");
$producer->start();

for ($i = 0; $i < 10000000; $i++) {
    try {
        $msg = new Message("TopicTest", 
        "TagA",
        "OrderID188",
        "Hello world");
        $sendResult = $producer->send($msg);
        echo $sendResult;
    } catch (\Exception $e) {
        echo $e->getMessage() . PHP_EOL . $e->getTraceAsString();
    }
}
$producer->shutdown();

File: 系列3-分布式思维及源码分析/31-Rocketmq特性详解-悟空/vip2018rocketmq/src/main/java/com/tuling/Producer.java
package com.tuling;

import org.apache.rocketmq.client.exception.MQBrokerException;
import org.apache.rocketmq.client.exception.MQClientException;
import org.apache.rocketmq.client.producer.DefaultMQProducer;
import org.apache.rocketmq.common.message.Message;
import org.apache.rocketmq.remoting.exception.RemotingException;

import java.util.UUID;

public class Producer {
    public static void main(String[] args) {
        //生成Producer
        DefaultMQProducer producer = new DefaultMQProducer("wukongGroup");
        //配置Producer
        producer.setNamesrvAddr("192.168.0.12:9876;192.168.0.13:9876");
        producer.setInstanceName(UUID.randomUUID().toString());
        //启动Producer
        try {
            producer.start();
        } catch (MQClientException e) {
            e.printStackTrace();
            return;
        }
        //生产消息
        String str = "悟空来了" + UUID.randomUUID().toString();
        Message msg = new Message("wukong", str.getBytes());
        try {
            producer.send(msg);
        } catch (MQClientException | RemotingException | MQBrokerException | InterruptedException e) {
            e.printStackTrace();
            return;
        }

        //停止Producer
        producer.shutdown();
        System.out.print("[----------]Succeed\n");
    }
}